images/multus-cni/tests/deploy.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+# Function to check pod status
+check_pod_status() {
+    local pod_name="$1"
+    local namespace="$2"
+    kubectl get pod "$pod_name" -n "$namespace" &>/dev/null
+}
+
+# Function to delete the pod if it exists
+delete_pod_if_exists() {
+    local pod_name="$1"
+    local namespace="$2"
+    if check_pod_status "$pod_name" "$namespace"; then
+        kubectl delete pod "$pod_name" -n "$namespace" &>/dev/null
+    fi
+}
+
+# Apply the Multus CNI DaemonSet manifest
+kubectl apply -f https://raw.githubusercontent.com/k8snetworkplumbingwg/multus-cni/master/deployments/multus-daemonset-thick.yml  &>/dev/null
```

Comment:
This manifest also looks to be referencing the original upstream image:

```bash
      containers:
        - name: kube-multus
          image: ghcr.io/k8snetworkplumbingwg/multus-cni:snapshot-thick
```

So we're not testing our image with the DaemonSet. It might be easier to do the helm install first in terraform, then as the next test, break into this shell script and do the validation? we've a few examples of that in other images. 

But whatever approach we take we need to make sure we're testing our image, and it looks like this is still hardcoding the upstream image in this manifest

Reply:
Is this comment from [images/multus-cni/tests/deploy.sh](https://github.com/chainguard-images/images/pull/2395/files/699a47ace2d5ed0419efb35365d4dd61371c7ef3#diff-0780c12e9742071ea52da1b61403f2b8e6cfb2f9256db7df067dcd2acee071d7)
```
spec:
  serviceAccountName: multus
  containers:
  - name: multus-test-container
    image: $IMAGE_NAME
```

I change it to this variable, and inside of values.yaml used 
```
containers:
      - name: kube-multus
        # crio support requires multus:latest for now. support 3.3 or later.
        image: "${data.oci_string.ref}"
```

Reply:
Added the multus-daemonset-thick.yml to multus-values.yaml using the $IMAGE_NAME variable into the test.